gp-includes/assets-loader.php

```diff
@@ -1,84 +1,94 @@
-	$scripts->add( 'gp-editor', $url . '/editor.js', array( 'gp-common', 'jquery-ui-tooltip' ), '20160329' );
-	$scripts->add( 'gp-glossary', $url . '/glossary.js', array( 'gp-common' ), '20160329' );
-	$scripts->add( 'gp-translations-page', $url . '/translations-page.js', array( 'gp-common' ), '20150430' );
-	$scripts->add( 'mass-create-sets-page', $url . '/mass-create-sets-page.js', array( 'gp-common' ), '20150430' );
+	// Register our standard scripts.
+	wp_register_script( 'tablesorter', $url . '/jquery.tablesorter.min.js', array( 'jquery' ), '1.10.4' );
+	wp_register_script( 'gp-common', $url . '/common.js', array( 'jquery' ), '20150430' );
+	wp_register_script( 'gp-editor', $url . '/editor.js', array( 'gp-common', 'jquery-ui-tooltip' ), '20160329' );
+	wp_register_script( 'gp-glossary', $url . '/glossary.js', array( 'gp-editor' ), '20160329' );
+	wp_register_script( 'gp-translations-page', $url . '/translations-page.js', array( 'gp-editor' ), '20150430' );
+	wp_register_script( 'gp-mass-create-sets-page', $url . '/mass-create-sets-page.js', array( 'gp-editor' ), '20150430' );
 }
 
-add_action( 'wp_default_scripts', 'gp_scripts_default' );
+add_action( 'init', 'gp_register_scripts' );
 
 /**
- * Here we abstract WordPress core's enqueuing functions because...
- * 1. We don't want to print scripts and styles that are meant for the WordPress theme
- * 2. GlotPress enqueues scripts and styles from its template files and if we do that
- *    with wp_enqueue_script() and wp_enqueue_style() WordPress complains that
- *    those functions should only be called inside of wp_enqueue_scripts()
+ * Enqueue one or more styles.
+ *
+ * @param string|array $handles A single style handle to enqueue or an array or style handles to enqueue.
  */
-
-function gp_enqueue_scripts() {
-	global $gp_enqueued_styles, $gp_enqueued_scripts;
-
-	if ( ! empty( $gp_enqueued_scripts ) ) {
-		foreach ( $gp_enqueued_scripts as $handle ) {
-			wp_enqueue_script( $handle );
-		}
+function gp_enqueue_style( $handles ) {
+	// Check to see if $handles is an array, if not, then we can make it one to simplify the next loop.
+	if ( ! is_array( $handles ) ) {
+		$handles = array( $handles );
 	}
 
-	if ( ! empty( $gp_enqueued_styles ) ) {
-		foreach ( $gp_enqueued_styles as $handle ) {
-			wp_enqueue_style( $handle );
-		}
+	// Loop through each handle we've been asked to enqueue.
+	foreach ( $handles as $handle ) {
+		// Store the handle name in the global array.
+		GP::$styles[] = $handle;
+
+		// Actually enqueue the handle via WordPress.
+		wp_enqueue_style( $handle );
 	}
 }
 
-add_action( 'wp_enqueue_scripts', 'gp_enqueue_scripts' );
-
-function gp_enqueue_style( $handle ) {
-	global $gp_enqueued_styles;
-
-	$gp_enqueued_styles[] = $handle;
-}
+/**
+ * Enqueue one or more script.
+ *
+ * @param string|array $handles A single script handle to enqueue or an array or enqueue handles to enqueue.
+ */
+function gp_enqueue_script( $handles ) {
```

Comment:
I can change it, but it's going to break a lot of backwards compatibility with other plugins and custom templates.  If we change it (and the other one) should we include a wrapper to the singular version for backwards compatibility?